core/base/inc/TVirtualMutex.h

```diff
@@ -101,6 +108,8 @@ class TLockGuard {
 #define R__LOCKGUARD_NAMED(name,mutex) TLockGuard _NAME2_(R__guard,name)(mutex)
 #define R__LOCKGUARD_UNLOCK(name) _NAME2_(R__guard,name).UnLock()
 #else
+//Note: no trailing semicolon needed below, so API is not exactly similar depending on flags
```

Comment:
Well, "API is not exactly similar depending on flags" isn't as relevant as "you can write the call identically, whether _REENTRANT or not". I.e. your new comment "Use with a trailing semicolon" above makes sure that this difference (one requires a semicolon and the other one happily accepts a semicolon) becomes irrelevant. Or what am I missing?

Reply:
True. I was just thinking of backward compatibility. Imagine there are people that wrote in old programs R__LOCKGUARD(myMutexObject) and one time they decide to activate p_thread, they will get a compilation error. (like happened to me). But it's true that it doesn't make sense on the first place to pass the object instead of the pointer, so I will just remove the comment.